src/components/chips/js/chipsDirective.js

```diff
@@ -0,0 +1,159 @@
+	 * 'chips'. This component can make use of an <input> element or an <md-autocomplete> element.
+	 *
+	 *
+	 * TODO(typotter):
+	 * Expand input controls:
+	 *   Support md-autocomplete
+	 *   plain <input> tag as child
+	 *   textarea input
+	 *   md-input??
+	 * List Manipulation
+	 *   delete item via DEL or backspace keys when selected
+	 * Validation
+	 *   de-dupe values (or support duplicates, but fix the ng-repeat duplicate key issue)
+	 *   allow a validation callback
+	 *   hilighting style for invalid chips
+	 * Item mutation
+	 *   Support <md-chip-edit> template, show/hide the edit element on tap/click? double tap/double click?
+	 * Truncation and Disambiguation (?)
+	 *   Truncate chip text where possible, but do not truncate entries such that two are indistinguishable.
+	 * Drag and Drop (?)
+	 *   Drag and drop chips between related <md-chips> elements.
+	 *
+	 *
+	 * @param {string=|object=} ng-model A model to bind the list of items to
+	 * @param {string=} placeholder Placeholder text that will be forwarded to the input.
+	 * @param {string=} secondary-placeholder Placeholder text that will be forwarded to the input, displayed when there
+	 *    is at least on item in the list
+	 * @param {boolean=} readonly Disables list manipulation (deleting or adding list items), hiding the input and delete
+	 *    buttons
+	 * @param {expression} md-chip-append An expression expected to convert the input string into an object when adding
+	 *    a chip.
+	 *
+	 * @usage
+	 * <hljs lang="html">
+	 *   <md-chips
+	 *       ng-model="myItems"
+	 *       placeholder="Add an item"
+	 *       readonly="isReadOnly">
+	 *   </md-chips>
+	 * </hljs>
+	 *
+	 */
+
+
+	var MD_CHIPS_TEMPLATE = '\
+			<md-chips-wrap ng-class="{readonly : $mdChipsCtrl.readonly}" class="md-chips">\
+				<div role="presentation">\
+					<div ng-repeat="$chip in $mdChipsCtrl.items"\
+							ng-class="{selected: $mdChipsCtrl.selectedChip == $index}"\
+							ng-click="!$mdChipsCtrl.readonly && $mdChipsCtrl.selectChip($index)"\
+							class="md-chip">\
+					</div>\
+					<div ng-if="!$mdChipsCtrl.readonly" class="md-chip-worker"></div>\
+				</div>\
+			</md-chips-wrap>';
+
+  var CHIP_INPUT_TEMPLATE = '\
+				<input\
+						placeholder="{{$mdChipsCtrl.items.length == 0 ? $mdChipsCtrl.placeholder : $mdChipsCtrl.secondaryPlaceholder}}"\
+						class="md-chip-input"\
+						ng-model="$mdChipsCtrl.chipBuffer"\
+						ng-focus="$mdChipsCtrl.resetSelectedChip()"\
+						ng-keydown="$mdChipsCtrl.defaultInputKeydown($event)">';
+
+  var CHIP_DEFAULT_TEMPLATE = '\
+			<md-chip>\
+				<span>{{$chip}}</span>\
+				<md-button ng-if="!$mdChipsCtrl.readonly"\
+						md-chip-remove>x</md-button>\
+			</md-chip>';
+
+
+  /**
+   * MDChips Directive Definition
+   * @param $mdTheming
+   * @param $log
+   * @ngInject
+   */
+  function MdChips ($mdTheming, $log) {
+    return {
+      template: function(element, attrs) {
```

Comment:
@jelbourn can probably elaborate on the trickery since he showed me this trick... The template property allows for a function instead of a string which takes the original HTML of the component and its attributes as parameters. Via Jeremy's trick, I am storing the original HTML in a hidden attribute so that I can do some of my own manual trans- and in-clusion of child elements. In this case, I am only including (i.e. the hunk of HTML I take from the caller and include in the final rendering takes on the scope of its destination). In a following change, I will be doing a poor-man's transclude of an md-autocomplete or input element, by $compiling its HTML with the component scope's parent scope.